### PR TITLE
Fix bug when compiling OpenPMD + 2D

### PR DESCRIPTION
**Pull request by MaxThevenet**
```sh
make -j 8 USE_OPENPMD=TRUE DIM=2
```
gave compilation errors, this PR should fix them.

### CODE REVIEW

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 1]
```diff
@@ -367,9 +367,9 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
         std::shared_ptr<amrex::ParticleReal> ptr_uz = ps.second["momentum"]["z"].loadChunk<amrex::ParticleReal>();
         double const momentum_unit_z = ps.second["momentum"]["z"].unitSI();
 #   else
-        amrex::Abort("AddPlasmaFromFile is only implemented for 2D and 3D\n")
+        amrex::Abort("AddPlasmaFromFile is only implemented for 2D and 3D\n");
 #   endif
-#   if (defined WARPX_DIM_3D)
+#   if (defined WARPX_DIM_XZ)
```
Comment (LDAmorim) [inline Source/Particles/PhysicalParticleContainer.cpp:372]: Just to double check, is this supposed to be XZ and not 3D?

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 2]
```diff
@@ -367,9 +367,9 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
         std::shared_ptr<amrex::ParticleReal> ptr_uz = ps.second["momentum"]["z"].loadChunk<amrex::ParticleReal>();
         double const momentum_unit_z = ps.second["momentum"]["z"].unitSI();
 #   else
-        amrex::Abort("AddPlasmaFromFile is only implemented for 2D and 3D\n")
+        amrex::Abort("AddPlasmaFromFile is only implemented for 2D and 3D\n");
```
Comment (LDAmorim) [inline Source/Particles/PhysicalParticleContainer.cpp:370]: Thanks for catching this!
And we will be able to remove this line after testing PR #956 in RZ :)

File: Source/Particles/PhysicalParticleContainer.cpp (hunk 1 above)
Comment (LDAmorim) [reply Source/Particles/PhysicalParticleContainer.cpp:372]:
```suggestion
#   if (defined WARPX_DIM_3D)
```

Comment (MaxThevenet) [reply Source/Particles/PhysicalParticleContainer.cpp]: Oops, absolutely, thanks for the catch.

**Comment by LDAmorim** [comment]
Would it be worth it to also (here or in other PRs) make consistent the entries with
```cpp
#if (AMREX_SPACEDIM==3)
```
?

File: Source/Particles/PhysicalParticleContainer.cpp (hunk 2 above)
Comment (ax3l) [reply Source/Particles/PhysicalParticleContainer.cpp:370]: I saw this and though: "that's probably a macro, no problem" xD Upsi!

**Comment by MaxThevenet** [comment]
@LDAmorim so far we're not too careful about that, so this could be fixed in a separate PR in many places in the code. While the distinction is clear between `AMREX_SPACEDIM == 2` and `WARPX_DIM_XZ` and `WARPX_DIM_RZ`, we still need all of these flags because they mean different things. For now, `AMREX_SPACEDIM == 3` and `WARPX_DIM_3D` are the same thing but there are several examples where they could mean different things, so I think it is fine to have both for now.

**Comment by LDAmorim** [comment]
Thank you for explaining @MaxThevenet !!
I understood that for now the two 3D ones seemed to be used for the same cases. But if they can mean something else in the future then it is great to keep both.

**Pull request merged**